shared-scripts.sh

```diff
@@ -132,6 +132,72 @@ function _lint {
   loadCacheFromBuildJob
   chmod +x codebuild_specs/scripts/lint_pr.sh && ./codebuild_specs/scripts/lint_pr.sh
 }
+function _verifyAmplifyBackendCompatability {
+  echo "Verify Amplify Backend Compatibility"
+  loadCacheFromBuildJob
+
+  # Unset container credentials environment variables since some of the tests in packages/cli/src/command_middleware.test.ts 
+  # expect not to fetch the credentials. This is to avoid the tests from failing.
+  echo "Unsetting container credentials environment variables"
+  unset AWS_CONTAINER_CREDENTIALS_RELATIVE_URI
+  unset AWS_CONTAINER_CREDENTIALS_FULL_URI
+
+  # 1. Install NVM and set up
```

Comment:
Since we're explicitly overriding the container's node version, add a quick note in this comment about why.